Add .gitignore and pi_local.sh.

jasondoze opened the pull request:
Add aws_install.sh for aws cli.

Add terraform_install.sh.

Add main.tf file and varibles.tf.

Add terraform.tfvars for aws keys to .gitignore.

Add azure_install.sh to azure directory.

jasondoze committed:
Add .gitignore and pi_local.sh.

Add aws_install.sh for aws cli.

Add terraform_install.sh.

Add main.tf file and varibles.tf.

Add terraform.tfvars for aws keys to .gitignore.

Add azure_install.sh to azure directory.

Signed-off-by: jason <dozejason@gmail.com>

stephenlauck commented on aws/main.tf:
```diff
@@ -0,0 +1,24 @@
+  required_version = ">= 1.2.0"
+}
+provider "aws" {
+    access_key = "${var.access_key}"
```
What's better var or Environment Variables?

jasondoze replied:
Trying to keep my secrets out of main.tf with variables. Appears folks keep secrets in terraform.tfvars so that’s what I did. I don’t know how this is supposed to be setup, just going from AWS terraform examples and trying to figure it out.

stephenlauck commented on aws/main.tf:
```diff
@@ -0,0 +1,24 @@
+}
+
+resource "aws_instance" "ec2_instance" {
+    ami = "${var.ami_id}"
```
Why are you using var here? Why not just put the values in the main? Is there some reason to have extra files?

stephenlauck commented on aws/terraform_install.sh:
```diff
@@ -0,0 +1,59 @@
+# This script updates the package list, installs Gnupg, downloads the Hashicorp signing key, adds the Hashicorp repository and installs Terraform.
+
+# Update Apt
+if (( $(date +%s) - $(stat -c %Y /var/lib/apt/periodic/update-success-stamp) > 24*60*60 ))
```
A comment on what this is would be cool.

jasondoze replied:
This checks if apt has been updated in the last 24 hours. Seems some debate on what actually gets cached whenever apt is updated. Idk if this is the way but it seemed to have a lot of support on SO.

stephenlauck commented on aws/terraform_install.sh:
```diff
@@ -0,0 +1,59 @@
+  https://apt.releases.hashicorp.com $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/hashicorp.list
+  sudo apt update
+fi
+
```
Could you check here with another conditional block if terraform is in the apt-cache and remove the apt update earlier in the script?

stephenlauck commented on aws/variables.tf:
```diff
@@ -0,0 +1,36 @@
+variable "access_key" {
```
What's the point of this file really?

jasondoze replied:
At first I had all my secrets in main.tf. I had main.tf in .gitignore, then saw the terraform docs breaking these up into individual files. Idkwtf im doing. Trying to figure it out though.

jasondoze committed:
Add comment to apt update conditional and add print last update.

Signed-off-by: jason <dozejason@gmail.com>

jasondoze committed:
Delete 2 files, azure_install and variables.tf.

Rename 3 files.

Remove secrets from main.tf.

Add .env file for secrets, add to .gitignore.

Add deploy and destroy scripts for terraform aws instance.

Add vm deploy and destroy scripts.

Move pi_local and install terraform to project root.

Signed-off-by: jason <dozejason@gmail.com>

jasondoze committed:
Add arguments to pass to vm_deploy for each instance.

Add deploy_azure terraform command script.

Add main.tf that creates an Azure instance without using variables.

It works!!

Signed-off-by: jason <dozejason@gmail.com>

stephenlauck commented on azure/deploy_azure.sh:
```diff
@@ -0,0 +1,27 @@
+# Deploy Azure instance using terraform
+
+# Export environment variables
+source .env
```
What does this do?

jasondoze replied:
Just an assertion to see if you're paying attention, haha.

stephenlauck replied:
ha

stephenlauck commented on azure/install_azure.sh:
```diff
@@ -2,26 +2,105 @@
-# create vm https://learn.microsoft.com/en-us/azure/virtual-machines/linux/quick-create-cli
\ No newline at end of file
+# Login to Azure
+az login --tenant $ARM_TENANT_ID
```
what's the assertion here?

stephenlauck commented on install_terraform.sh:
```diff
@@ -17,7 +17,7 @@ then
 else
   echo -e "\n\033[1;33m==== Installing Gnupg ====\033[0m\n"
-  install -y gnupg software-properties-common
+  sudo apt install -y gnupg software-properties-common
```
break it up?

stephenlauck commented on vm_deploy.sh:
```diff
@@ -75,8 +75,23 @@ fi
+if [ "$1" == "aws" ]
+then
+  echo -e "\n\033[1;32m==== Running AWS scripts ====\033[0m\n"
+  ssh -o StrictHostKeyChecking=no -i ./id_ed25519 $USER@$(multipass info nameless | grep IPv4 | awk '{ print $2 }') 'cd multicloud_terraform && bash install_terraform.sh && cd aws && bash install_aws.sh && bash deploy_aws.sh'
```
instead of using IF statements, maybe checkout CASE statements. I think you may just want to make a variable called $CLOUD and just reference that like:

`cd multicloud_terraform && bash install_terraform.sh && cd $CLOUD && bash install_$CLOUD.sh && bash deploy_$CLOUD.sh`

jasondoze committed:
Add case conditional to pass arguments to install scripts.

Add assertion for Azure login in install_azure.

Remove source .env from deploy aws and azure files.

Add assertions for gnupg and package dependencies in install_terraform.sh

Add skeleton for main.tf in gcp directory.

Signed-off-by: jason <dozejason@gmail.com>
